File: golang/118/118.go
package main

func generate(numRows int) [][]int {
	trian := [][]int{}
	if numRows == 0 {
		return trian
	}
	trian = append(trian, []int{1})
	for i := 1; i < numRows; i++ {
		tmp := []int{1}
		for j := 1; j < i; j++ {
			tmp = append(tmp, trian[i-1][j-1]+trian[i-1][j])
		}
		tmp = append(tmp, 1)
		trian = append(trian, tmp)
	}
	return trian
}

File: cpp/118.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res;
        if(numRows == 0)  return res;
        res.push_back(vector<int>{1});
        for(int i=1; i<numRows; i++){
            vector<int> vec{1};
            for(int j=1; j<i; j++){
                vec.push_back(res[i-1][j-1] + res[i-1][j]);
            }
            vec.push_back(1);
            res.push_back(vec);
        }
        return res;
    }
};